rules.py

```diff
@@ -912,6 +912,47 @@ def check(self):
                 "The Election File has following issues with the identifiers.", error_log)
 
 
+class AllCaps(base.BaseRule):
+    """The Name elements in Candidates, Contests and Person elements should not be in all uppercase.
+
+    If the name elements in Candidates, Contests and Person elements are in uppercase, 
+    the list of bojectIds of those elements will be returned to the user as a warning."""
+
+    def elements(self):
```

Comment:
The `elements()` function from some of the other checks may be better examples, actually.

E.g.,
https://github.com/google/election_results_xml_validator/blob/master/rules.py#L205
https://github.com/google/election_results_xml_validator/blob/master/rules.py#L223
https://github.com/google/election_results_xml_validator/blob/master/rules.py#L238
https://github.com/google/election_results_xml_validator/blob/master/rules.py#L481

You should be able to just specify a list with Candidate, Contest, and Person.